### PR TITLE
Check for invariant culture before matching regex.

### PR DESCRIPTION
**Pull request by Priya91**
cc @tarekgh 

potential fix for #10400

### CODE REVIEW

File: src/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs [hunk 1]
```diff
@@ -613,12 +613,16 @@ public static IEnumerable<object[]> Groups_CustomCulture_TestData()
...
+                // In invariant culture, the unicode char matches differ from expected values provided.
+                if (originalCulture == CultureInfo.InvariantCulture)
+                {
+                    CultureInfo.CurrentCulture = new CultureInfo("en-US");
+                }
                 if (cultureInfo != null)
                 {
-                    originalCulture = CultureInfo.CurrentCulture;
                     CultureInfo.CurrentCulture = cultureInfo;
                 }
```
Comment (tarekgh) [inline src/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs]: do we need to ensure cultureInfo is not invariant too?

Comment (Priya91) [inline src/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs]: hmm but if the dev is setting cultureinfo to invariant, then that's a conscious decision on what to test for.. Changing that behavior is not what we want..

Comment (tarekgh) [inline src/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs]: right but the test will fail again. if we allow invariant culture to be passed then we should have the test written the way will not fail.

Comment (Priya91) [inline src/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs]: Yeah, but when invariant culture is passed the actual and expected values will also change, those values are specified here ```Groups_Basic_TestData```

Comment (tarekgh) [inline src/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs]: got it. the change LGTM. thanks.

---
In reply to: [74858742](https://github.com/dotnet/corefx/pull/10806#discussion_r74858742) [](ancestors = 74858742)

**Comment by Priya91** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test

**Comment by Priya91** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test

**Comment by Priya91** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Debug Build and Test

**Pull request merged**